mcs/tools/linker/Makefile

```diff
@@ -2,21 +2,40 @@ thisdir = tools/linker
 SUBDIRS =
 include ../../build/rules.make
 
-RESOURCES = \
-	Descriptors/mscorlib.xml	\
-	Descriptors/System.xml		\
-	Descriptors/System.Core.xml		\
-	Descriptors/System.Drawing.xml	\
-	Descriptors/System.Web.xml		\
-	Descriptors/Mono.Posix.xml		\
-
 PROGRAM = monolinker.exe
 
-$(PROGRAM): $(RESOURCES)
-
 LIB_REFS = System System.Core System.Xml Mono.Cecil
-LOCAL_MCS_FLAGS = $(RESOURCES:%=-resource:%)
 
-EXTRA_DISTFILES = $(RESOURCES)
+TEST_CASES = \
+	mscorlib/test-array.cs
+
+ifndef AOT_FRIENDLY_PROFILE
+TEST_CASES += \
+	mscorlib/test-reflection.cs
+endif
+
+TESTS_COMPILER = $(MCS) -nologo -noconfig -debug:portable -r:$(topdir)/class/lib/$(PROFILE)/mscorlib.dll
+
+check: compile-tests run-tests
+
+compile-tests: $(TEST_CASES)
+
+mscorlib/test-%.cs:
+	$(TESTS_COMPILER) Tests/$@ /out:Tests/$(@:.cs=.exe)
+
+run-tests: $(TEST_CASES:.cs=.exe)
+
+LINKER_OUTPUT := illink-output-$(PROFILE)
+PROFILE_PATH = $(topdir)/class/lib/$(PROFILE)
+LINKER = MONO_PATH=$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE) $(RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/monolinker.exe -c link -out $(LINKER_OUTPUT) -b true -d $(PROFILE_PATH)
+TEST_EXEC = MONO_PATH=$(LINKER_OUTPUT) $(RUNTIME) $(RUNTIME_FLAGS) --debug -O=-aot
```

Comment:
Because that's default linker behaviour, you can choose to regenerate but I was to test the worst case